### ISSUE TITLE
Reuse CMS component in multiple components with same endpoint

### ISSUE THREAD

**Issue by TommySorensen** [shuding/next-cms#8]
**Question: Is it possible to reuse the data the CMS component returns in multiple components with this way of fetching data?**

I am looking for a way to get the fetched data from getStaticProps down to multiple components. And it seems like just nesting the values from getStaticProps down 4 levels is a bit ugly when i need the data 50 places. Your solution in this repo seems like is a smart way to add multiple endpoints up to the staticProps.

So, will i be able to use the same endpoint multiple places?
Not exactly like this, but you get the point.
```js
import Link from "next/link";
import cn from "classnames";

import CMS from "../lib/cms";

import styles from "../styles/post.module.css";

export default function PostPreviews() {
  return (
    <>
      <CMS endpoint="/wp-json/wp/v2/posts?_fields=id,excerpt,title,date">
        {(posts) => posts.map((p) => <div>do something with data</div>)}
      </CMS>

      <CMS endpoint="/wp-json/wp/v2/posts?_fields=id,excerpt,title,date">
        {(posts) => posts.map((p) => <div>do something with data</div>)}
      </CMS>
    </>
  );
}
```

**Comment by shuding** [comment]
Yep, same endpoint will always be deduplicated and the data will be reusable.

**Comment by TommySorensen** [comment]
> Yep, same endpoint will always be deduplicated and the data will be reusable.

Thx. Will you suggest this as a way to get generic data from a CMS down to 100+ places in an app or does this seem a bit silly and overkill? I haven't found a good example for generic translations from a CMS yet 🤔

**Comment by shuding** [comment]
Yep I think it will still work fine. But you definitely need some pressure testing before putting it into production so you can add proper rate limiting for example.

**Comment by TommySorensen** [comment]
@shuding Was wondering if this example of how to fetch stuff to getStaticProps dynamically is still something you would recommend? I haven't found a better example of how it can be done 🙂

**Comment by shuding** [comment]
@TommySorensen Yep still no alternative way. In the near future I can see Suspense based SSR/SSG solving this but it’s not ready yet.
